### PR TITLE
Automation gap - WOAUILIB-3704

### PR DESCRIPTION
**Pull request by NitzanWix**
## Description
Added automation gap for text alignment on rtl.

## Changelog
N/A

## Additional info
WOAUILIB-3704

### CODE REVIEW

File: src/components/text/__tests__/index.driver.spec.tsx [hunk 1]
```diff
@@ -44,3 +45,45 @@ describe('Text', () => {
     });
   });
 });
+
+// jest.mock('react-native/Libraries/ReactNative/I18nManager', () => ({
```
Comment (ethanshar) [inline src/components/text/__tests__/index.driver.spec.tsx:49]: @NitzanWix 
Instead of mocking RN it might be simpler to override/mock our Constants.isRLT value
See here
https://github.com/wix/react-native-ui-lib/blob/master/src/components/sortableGridList/__tests__/usePresenter.spec.tsx

Comment (M-i-k-e-l) [reply src/components/text/__tests__/index.driver.spec.tsx:49]: He's testing the style here, so I don't think that'll work.
Although I'm not sure this is the best way to test this bug, an alternative might be with screenshots and I think we don't want to rely on that here.

Comment (NitzanWix) [reply src/components/text/__tests__/index.driver.spec.tsx:49]: I'm trying to test the following. overriding/mocking in Constants will work for RTL but I want to also mock the platform. Currently this doesn't work and pops a invalid hook usage error.
https://github.com/wix/react-native-ui-lib/blob/068b4d00174bbc13f1ef33dfcb811d9ab7dee369/src/components/text/index.tsx#L195-L203

File: src/components/text/__tests__/index.driver.spec.tsx [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
-import Text from '../index';
 import {TextDriver} from '../Text.driver.new';
+import {StyleSheet} from 'react-native';
 
 const TEXT_ID = 'text_test_id';
 const TEXT_CONTENT = 'text content';
 
 function WrapperScreenWithText(textProps: {onPress?: jest.Mock} = {}) {
   const {onPress} = textProps;
+  const Text = require('../../text').default;
```
Comment (M-i-k-e-l) [inline src/components/text/__tests__/index.driver.spec.tsx:11]: Why did you move this here?

File: src/components/text/__tests__/index.driver.spec.tsx [hunk 3]
```diff
@@ -44,3 +42,30 @@ describe('Text', () => {
...
+  Constants.isIOS = !isAndroid;
+  Constants.isRTL = isRTL;
+};
+
+describe('Automation gap', () => {
+  it('Should render text on right on rtl - Android', () => {
+    jest.isolateModules(() => {
+      setConstants(true, true);
+      const renderTree = render(<WrapperScreenWithText/>);
+      const textDriver = TextDriver({renderTree, testID: TEXT_ID});
```
Comment (M-i-k-e-l) [inline src/components/text/__tests__/index.driver.spec.tsx:57]: Consider adding `getDriver`

File: src/components/text/index.tsx [hunk 4]
```diff
@@ -192,14 +191,13 @@ const styles = StyleSheet.create({
   container: {
     backgroundColor: 'transparent',
     color: Colors.$textDefault,
-    ...Platform.select({
-      ios: {
+    ...(Constants.isIOS
```
Comment (M-i-k-e-l) [inline src/components/text/index.tsx:194]: Nice idea!
Did you verify this change works (in the app, not tests)

File: src/components/text/__tests__/index.driver.spec.tsx [hunk 5]
```diff
@@ -44,3 +42,30 @@ describe('Text', () => {
...
   });
 });
+const setConstants = (isAndroid: boolean, isRTL: boolean) => {
+  const {Constants} = require('../../../commons/new');
+  Constants.isAndroid = isAndroid;
+  Constants.isIOS = !isAndroid;
+  Constants.isRTL = isRTL;
+};
+
+describe('Automation gap', () => {
```
Comment (M-i-k-e-l) [inline src/components/text/__tests__/index.driver.spec.tsx:52]: Consider adding tests for LTR as well

File: src/components/text/__tests__/index.driver.spec.tsx (hunk 2 above)
Comment (NitzanWix) [reply src/components/text/__tests__/index.driver.spec.tsx:11]: Because the styling is being set when Text is loaded (in StyleSheet.create) I needed a way to make it load again each time I test. Requiring it there and using jest.isolateModules let me do that because its not being cached this way. This is the Same reason for requiring Constants inside setConstants, because Text uses it it loads it before I can change it so this way by loading it before Text does and changing its values Text uses the appropriate values.

File: src/components/text/index.tsx (hunk 4 above)
Comment (NitzanWix) [reply src/components/text/index.tsx:194]: Seems to be working

File: src/components/text/__tests__/index.driver.spec.tsx (hunk 5 above)
Comment (NitzanWix) [reply src/components/text/__tests__/index.driver.spec.tsx:54]: For text? Or in general?

Comment (M-i-k-e-l) [reply src/components/text/__tests__/index.driver.spec.tsx:54]: The same tests but with LTR

File: src/components/text/__tests__/index.driver.spec.tsx [hunk 6]
```diff
@@ -1,46 +1,71 @@
...
+      {TEXT_CONTENT}
+    </Text>
   );
 }
 
+const getDriver = (textProps: {onPress?: jest.Mock} = {}) => {
+  const {onPress} = textProps;
+  const renderTree = render(<WrapperScreenWithText onPress={onPress}/>);
+  const textDriver = TextDriver({renderTree, testID: TEXT_ID});
+  return {renderTree, textDriver};
```
Comment (M-i-k-e-l) [inline src/components/text/__tests__/index.driver.spec.tsx:23]: Don't think you need to return `renderTree` here

**Review by M-i-k-e-l — APPROVED** [review]
Nice

File: src/components/text/__tests__/index.driver.spec.tsx [hunk 7]
```diff
@@ -1,46 +1,91 @@
...
     });
   });
+  const setConstants = (isAndroid: boolean, isRTL: boolean) => {
+    const {Constants} = require('../../../commons/new');
+    Constants.isAndroid = isAndroid;
+    Constants.isIOS = !isAndroid;
+    Constants.isRTL = isRTL;
+  };
+
+  describe('Automation gap', () => {
```
Comment (Inbal-Tish) [inline src/components/text/__tests__/index.driver.spec.tsx:53]: Please change the name
